libs/pika/futures/src/future_data.cpp

```diff
@@ -125,7 +125,8 @@ namespace pika { namespace lcos { namespace detail {
         if (s == empty)
         {
             // the value has already been moved out of this future
-            PIKA_THROWS_IF(ec, no_state, "future_data_base::get_result",
+            PIKA_THROWS_IF(ec, pika::error::future_already_retrieved,
```

Comment:
seems better thanks!

Reply:
This was actually a mistake :sweat_smile: But it makes sense here. I've changed some other instances of `future_already_retrived` back to `no_state` though (as they are currently on main).